### DISCUSSION TITLE
Selenium 4.0.0 Compatibility

### DISCUSSION THREAD

**Discussion by faidootdoot** [TroyWalshProf/SeleniumAxeDotnet#208] [Q&A] [answered]
Is this Selenium.Axe 3.0.1 compatible with Selenium 4x?

Selenium.Axe 3.0.1 successfully created html reports using the last stable version of Selenium 3.141.
However, using Selenium 4.0.0-rc1, on calling CreateAxeHtmlReport an error is generated.

  Message: 
    System.TypeLoadException : Could not load type 'OpenQA.Selenium.Internal.IWrapsElement' from assembly 'WebDriver, Version=4.0.0.0, Culture=neutral, PublicKeyToken=null'.

  Stack Trace: 
    HtmlReport.CreateAxeHtmlReport(ISearchContext context, AxeResult results, String destination, ReportTypes requestedResults)

Thanks.

**Comment by TroyWalshProf** [comment]
Tested it out and tried to reproduce.
The result was Selenium.Axe 3.0.1 worked with Selenium 4.0.0‑rc1 and 4.0.0‑rc2.

Please provide more info so I can track this done:
Could I get the full stack trace?
Could I get the web driver setup and CreateAxeHtmlReport call code?
What other package do you depend on? 
Any way I could get the list of your other dependencies and dependency version?

**Reply by TroyWalshProf** [reply]
@faidootdoot - Not sure why I was able to repo this before but absolutely see the issue now.
This will be fixed in https://github.com/TroyWalshProf/SeleniumAxeDotnet/pull/209

**Comment by TroyWalshProf** [comment, marked as answer]
This has been fixed in Selenium.Axe 3.1

**Reply by faidootdoot** [reply]
Apologise for not getting back to you sooner.
Thank you very much for the quick turn around on my post.
I have updated accessibility tests to use Selenium.Axe 3.1 and Selenium 4 and html reports and json files are all happily generated.
